

On Wednesday, May 30, 2001 03:03:32 PM -0600 "D. Stimits"
<stimits@idcomm.com> wrote:

[ snip ]

> RAMDISK: Compressed image found at block 0
> Freeing initrd memory: 249k freed
> VFS: Mounted root (ext2 filesystem).
> Red Hat nash version 3.0.10 starting
> VFS: Mounted root (ext2 filesystem) readonly.
> change_root: old root has d_count=2
> Trying to unmount old root ... <1>Unable to handle kernel NULL pointer
> dereference at virtual address 00000010
>  printing eip:

Can't say for sure without the oops decoded through ksymoops, but this
looks like the oops in rd_ioctl fixed by 2.4.5-ac3 and higher.  I think the
following patch (taken from ac3) will be sufficient:

-chris

--- linux.vanilla/fs/block_dev.c	Sat May 26 16:53:17 2001
+++ linux.ac/fs/block_dev.c	Mon May 28 16:10:59 2001
@@ -603,6 +602,7 @@
 	if (!bdev->bd_op->ioctl)
 		return -EINVAL;
 	inode_fake.i_rdev=rdev;
+	inode_fake.i_bdev=bdev;
 	init_waitqueue_head(&inode_fake.i_wait);
 	set_fs(KERNEL_DS);
 	res = bdev->bd_op->ioctl(&inode_fake, NULL, cmd, arg);

-
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/
